plugin-management-library/src/main/java/io/jenkins/tools/pluginmanager/util/PluginManagerUtils.java

```diff
@@ -17,6 +24,12 @@
 
 public final class PluginManagerUtils {
 
+    /**
+     * The list of possible repository path prefixes used to resolve the update center URL.
+     * These prefixes are combined with the cache suffix to form the full path to the update center JSON file.
+     */
+    private static final String[] DEFAULT_ARCHIVE_REPOS_PATH_PREFIXES  = { "stable", "dynamic", "dynamic-stable" };
```

Comment:
this should be implementation details and we shouldn't be looking at this. The update center metadata should be providing us with a URL.

For archives it should just be a list of all plugins and versions under here: 
https://archives.jenkins.io/plugins/